contracts/multisig-improved/src/action_types/execute_action.rs

```diff
@@ -13,25 +15,49 @@ pub const MAX_BOARD_MEMBERS: usize = 30;
 
 #[multiversx_sc::module]
 pub trait ExecuteActionModule:
-    crate::state::StateModule
+    crate::common_functions::CommonFunctionsModule
+    + crate::state::StateModule
+    + super::external_module::ExternalModuleModule
     + crate::external::events::EventsModule
     + crate::ms_endpoints::callbacks::CallbacksModule
 {
+    fn try_execute_deploy(
+        &self,
+        action_id: ActionId,
+        action: &Action<Self::Api>,
+    ) -> OptionalValue<ManagedAddress> {
+        if let Action::SCDeployFromSource(args) = action {
+            let new_address = self.deploy_from_source(action_id, args.clone());
+
+            return OptionalValue::Some(new_address);
+        }
+        if let Action::DeployModuleFromSource(args) = action {
```

Comment:
We should not deploy them from here - we should only add them by address. The addition should go through a proposal and the only checks we need is that it's a deployed sc and it has the can_execute func if possible

Reply:
Also same-shard check